File: 02_ProgrammingFundamentals/_04_Methods/_02_Exercise/_06_MiddleCharacters.java
package _02_ProgrammingFundamentals._04_Methods._02_Exercise;

import java.util.Scanner;

public class _06_MiddleCharacters {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        // Read input value
        String console = scan.nextLine();

        // Create method that prints the middle of a string
        middleString(console);

        scan.close();
    }

    private static void  middleString(String console) {
        if (console.length() % 2 == 0) {
            char middleEvenOne = console.charAt((console.length() / 2) - 1);
            char middleEvenTwo = console.charAt(console.length() / 2);

            System.out.printf("%c%c", middleEvenOne, middleEvenTwo);
        } else {
            char middleOdd = console.charAt(console.length() / 2);

            System.out.println(middleOdd);
        }
    }
}

File: 24_C#Fundamentals/_04_Methods/_02_Exercise/_06_MiddleCharacters/_06_MiddleCharacters.cs
using System;

public class _06_MiddleCharacters
{
    public static void Main(string[] args)
    {
        // Read input value
        string console = Console.ReadLine();

        // Create method that prints the middle of a string
        middleString(console);
    }

    private static void middleString(string console)
    {
        if (console.Length % 2 == 0)
        {
            char middleEvenOne = console[(console.Length / 2) - 1];
            char middleEvenTwo = console[console.Length / 2];

            Console.WriteLine($"{middleEvenOne}{middleEvenTwo}");
        }
        else
        {
            char middleOdd = console[console.Length / 2];

            Console.WriteLine(middleOdd);
        }
    }
}
